src/main/webapp/js/all-section-deals.js

```diff
@@ -62,6 +77,63 @@ function initAllDeals() {
   }
 }
 
+/**
+ * Retrieves user's location
+ * @return {boolean} prevents the href from loading
```

Comment:
this boolean is very random, like it's not related to getting the location. Maybe not make it return a boolean, then in your onclick event, just return false?